src/coreclr/jit/assertionprop.cpp

```diff
@@ -2205,6 +2205,13 @@ AssertionInfo Compiler::optAssertionGenJtrue(GenTree* tree)
     GenTree* op1 = relop->AsOp()->gtOp1->gtCommaStoreVal();
     GenTree* op2 = relop->AsOp()->gtOp2->gtCommaStoreVal();
 
+    // Avoid creating local assertions for float types.
+    //
+    if (optLocalAssertionProp && varTypeIsFloating(op1))
```

Comment:
👍. 

Just noting we could expand this in the future...

We have some well-defined APIs like `IsNaN`, `IsInfinity`, `IsFinite`, etc. These can all be used to drive other assertions.

Likewise if we know that op2 isn't NaN and isn't zero, then bitwise equality does hold (we take advantage of this in import to optimize `min` and `max` for example, when one input is a constant).

Reply:
Added a note to https://github.com/dotnet/runtime/issues/93246 so we don't forget about this.